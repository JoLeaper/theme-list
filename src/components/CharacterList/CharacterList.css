@import '../../styles/variable.css';

.CharacterList {
  display: grid;
  grid-template-columns: 1fr 5fr;
  transition: 0.2s;

  img {
    width: 10rem;
  }
}

.CharacterList-light {
  composes: CharacterList;

  background-color: var(--light-bg-color);
  color: var(--light-fg-color);
}

.CharacterList-dark {
  composes: CharacterList;

  background-color: var(--dark-bg-color);
  color: var(--dark-fg-color);
}
