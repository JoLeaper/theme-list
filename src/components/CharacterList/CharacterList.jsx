import React from 'react'
import PropTypes from 'prop-types'
import Character from '../Character/Character'

function CharacterList({ characterList }) {
  const CharacterListElements = characterList.map(character => <Character character={character} />);

  return (
    <ul>
      {<CharacterListElements />}
    </ul>
  )
}

CharacterList.propTypes = {
  characterList: PropTypes.array
}

export default CharacterList

