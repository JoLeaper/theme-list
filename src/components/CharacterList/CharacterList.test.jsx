import React from 'react';
import { shallow } from 'enzyme';
import CharacterList from './CharacterList';

describe('CharacterList component', () => {
  const characterList = [
    {
      id: 1,
      name: 'aang',
      image: 'testpicture.com'
    },
    {
      id: 2,
      name: 'sokka',
      image: 'testpicture.com'
    },
    {
      id: 3,
      name: 'katara',
      image: 'testpicture.com'
    },
    {
      id: 4,
      name: 'toph',
      image: 'testpicture.com'
    },
  
  ]

  it('renders CharacterList', () => {
    const wrapper = shallow(<CharacterList characterList={characterList}/>);
    expect(wrapper).toMatchSnapshot();
  });
});