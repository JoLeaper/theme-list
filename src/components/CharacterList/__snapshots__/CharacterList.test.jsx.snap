// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterList component renders CharacterList 1`] = `
<ul>
  <[object Object],[object Object],[object Object],[object Object] />
</ul>
`;
